improved plugins multilingual, customarchive

stephanbrunker opened the pull request:
see individual changelog files for details

event_customarchive:
   * The standard link to the archive /pages/archives.html causes the default 
     builtin archive to be displayed beneath the customarchive. This is
     caused by the matching in index.php with the 'archives' keyword.
     To disable that, change it to something else. It also does work with
     a relative link and without the .html like /sortarchiv
   * added a sidebar plugin instead of the html nugget. The plugin also 
     uses the relative path and shows an option to display a link to
     the builtin archive if it is not displayed underneath.
   * made the plugin multilingual: added a field for language-based-search,
     check for the settings of multilingual plugin (if installed), displays only
     the entries in the selected language if that option is set and strips the
     tagged translations from the categories
   * displays now a message if the result was empty
   * reformated the search mask

event_multilingual:
   * Rework of the language determination function, most of it now covered in
     the serendipity framework, details in leading comments and comments in
     /include/functions_config.inc.php: serendipity_getSessionLanguage() and
     serendipity_getPostAuthSessionLanguage(), added $serendipity['default_lang']
     for the blogs default language
   * activated and integrated the option to hide untranslated articles
   * changed the footer link to the format id-title for all languages,
     including default
   * The backend display language is now the user language, the dropdown
     entry for 'default' is fixed to the blog default language (has to be).
   * sidebar plugin: changed the POST value to [serendipity][user_language]
     to get it into $serendipity['POST']
   * added event_hook 'multilingual_strip_langs' to strip tagged translation from strings
     i. e. category use outside the sidebar

event_staticpage:
   * only fixed the title to the name of the plugin

event_entrypaging:
   * made the link matching to the selected language

th-h commented:
As far as I see we can merge the changes to multilingual only after the release of 2.4, as they depend on the changes in core, correct?

Are the changes to customarchive, staticpage and entrypaging dependent on the changes to multilingual? If not, it would be very good to split them off to their own PRs - it's not just a good idea in general to keep isolated changes separated, but it would make it possible to merge them sooner.

stephanbrunker commented:
The changes in the core doesn't depend on the multilingual plugin. It would be bad style if this would be the case. I just added eventhooks into the core and if the plugin doesn't listen, nothing happens ;-).

As far as it goes the other way round: because I added a variable $serendipity['default_lang'] which is used by the plugin, I think it would be a bad idea to release the plugin before 2.4 . In this case, it would be also neccessary to write this into the requirements section.

th-h commented:
> As far as it goes the other way round: because I added a variable $serendipity['default_lang'] which is used by the plugin, I think it would be a bad idea to release the plugin before 2.4 

Yes, that's what I meant.

stephanbrunker commented:
You have a point here and it was an easy fix. There is now an empty check for the variable and that makes it effectively downward compatible with previous versions. But I would strongly recommend to use 2.4 because half of the language fixes I had to place in the core.
